go/cmd/dolt/dolt.go

```diff
@@ -581,6 +582,14 @@ func buildLateBinder(ctx context.Context, cwdFS filesys.Filesys, mrEnv *env.Mult
 		targetEnv = mrEnv.GetEnv(useDb)
 	}
 
+	// There is no target environment detected. Return a cli.LateBindQueryist that returns an error.
+	// This stops commands that need to run queries, while allowing other commands to run.
+	if targetEnv == nil {
+		return func(ctx context.Context) (cli.Queryist, *sql.Context, func(), error) {
+			return nil, nil, nil, fmt.Errorf("The current directory is not a valid dolt repository.")
+		}, nil
+	}
+
 	// nil targetEnv will happen if the user ran a command in an empty directory - which we support in some cases.
```

Comment:
This line seems to indicate that nil targets are sometimes expected. I'm pretty sure this is what happens when you are in a directory that `dolt sql-server` has started on, but no databases exist yet, so I don't think that the fix above is what you want in all cases.